logstash-core/lib/logstash/instrument/snapshot.rb

```diff
@@ -14,8 +14,8 @@ def initialize
     end
 
     def push(*args)
-      type, key, _ = args
-      metric = @metrics.fetch_or_store(key, concrete_class(type, key))
+      namespace, key, type, _ = args
+      metric = @metrics.fetch_or_store([namespace, key].join('-'), concrete_class(type, key))
```

Comment:
This is an expensive call yes, I wanted to change the collection of the metric without breaking all the thing, so this was needed to reduce the size of this PR.

One way we could optimize this would be to use an hash map with precompiled keys?